cmd/iam-object-store.go

```diff
@@ -604,75 +530,50 @@ type itemOrErr struct {
 // prefix. If dirs is true, only directories are listed, otherwise
 // only objects are listed. All returned items have the pathPrefix
 // removed from their names.
-func listIAMConfigItems(objectAPI ObjectLayer, pathPrefix string, dirs bool,
-	doneCh <-chan struct{}) <-chan itemOrErr {
-
+func listIAMConfigItems(ctx context.Context, objAPI ObjectLayer, pathPrefix string, dirs bool) <-chan itemOrErr {
 	ch := make(chan itemOrErr)
-	dirList := func(lo ListObjectsInfo) []string {
-		return lo.Prefixes
-	}
-	filesList := func(lo ListObjectsInfo) (r []string) {
-		for _, o := range lo.Objects {
-			r = append(r, o.Name)
-		}
-		return r
-	}
 
 	go func() {
 		defer close(ch)
 
-		marker := ""
-		for {
-			lo, err := objectAPI.ListObjects(context.Background(),
-				minioMetaBucket, pathPrefix, marker, SlashSeparator, maxObjectList)
-			if err != nil {
-				select {
-				case ch <- itemOrErr{Err: err}:
-				case <-doneCh:
-				}
-				return
-			}
+		// Allocate new results channel to receive ObjectInfo.
+		objInfoCh := make(chan ObjectInfo)
 
-			// Attempt a slow down load only when server is
-			// active and initialized.
-			if !globalSafeMode {
-				// Slow down listing and loading for config items to
-				// reduce load on the server
-				waitForLowHTTPReq(int32(globalEndpoints.NEndpoints()))
+		// Walk through all objects
+		if err := objAPI.Walk(ctx, minioMetaBucket, pathPrefix, objInfoCh); err != nil {
```

Comment:
Walk() is a recursive listing but we don't want that (this is actually causing some issues while testing)

Reply:
This I will revert this is wrong @vadmeste